### PR TITLE
Create 2017-09-12-event-driven-serverless-app-local-dev-exp.md

### PR DESCRIPTION
**Pull request by rupakg**
[x] Add front matter yaml for post
[ ] Add thumbnail for post
[x] Add post content
[x] Add relevant images linked to S3 (if any)

### CODE REVIEW

File: posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+
+**Behind the scenes**
+ 
+Actually, there are a couple of things that the Serverless Application Platform does behind the scenes. The `serverless.yml` is parsed, and the `register` function from the `users` service is registered with the Event Gateway. Next, a subscription is created for the `http` event and the `register` function.
+
+We will visualize this when we run the service and talk about it, in the next section.
+
+Now, let's look at the handler code:
+
+```bash
```
Comment (DavidWells) [inline posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md]: use js syntax highlighter chere

File: posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+### Building the Email Service
+
+The workflow of the email service is:
+
+- send a welcome email to the registered user, and
+- **emit** an event `email.sent`
+
+Let's look at the functions section of the `serverless.yml`:
+
+```bash
```
Comment (DavidWells) [inline posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md]: use yaml syntax highlighter here

File: posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+
+**Behind the scenes**
+ 
+There are a couple of things that the Serverless Application Platform does behind the scenes. The `serverless.yml` is parsed, and the `sendWelcomeEmail` function from the `email` service is registered with the Event Gateway. Next, a subscription is created for the `user.registered` event and the `sendWelcomeEmail` function.
+
+We will visualize this when we run the service and talk about it, in the next section.
+
+Now, let's look at the handler code, specifically the `sendWelcomeEmail` function:
+
+```bash
```
Comment (DavidWells) [inline posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md]: js highlight

File: posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+{"id":26,"session":"e40e0d06ed05f17385dee72a56cfda48","email":"yourname@mail.com","name":"Rupak Ganguly"}%
+```
+
+![](https://s3-us-west-2.amazonaws.com/assets.blog.serverless.com/blog_app_sls_run_curl.gif)
+
+Let's break it up and discuss what just happened.
+
+1. Event Gateway receives the `http` event `POST http://localhost:4000/users`
+
+    ```bash
```
Comment (DavidWells) [inline posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md]: json

Comment (DavidWells) [reply posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md]: or maybe not. Up to you

**Comment by rupakg** [comment]
@DavidWells Thanks. Fixed.

File: posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+
+Here the Serverless Framework detects the `register` function in the `users` service and deploys it to the local Emulator.
+
+Then the `register` function from the `users` service is registered with the Event Gateway. And, a subscription is created for the `http` event and the `register` function.
+
+### Running the Email Service
+
+Let's run the email service now.
+
+![](https://s3-us-west-2.amazonaws.com/assets.blog.serverless.com/blog_app_sls_run_email.gif)
```
Comment (alexdebrie) [inline posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md]: This gif isn't showing for me.

Comment (rupakg) [reply posts/2017-09-12-event-driven-serverless-app-local-dev-exp.md]: yeah, it shows up when I copied it to a browser. let's publish and see if it works, or we will fix it.

**Comment by rupakg** [comment]
@alexdebrie schedule for Tue.

**Pull request merged**